src/mono/mono/metadata/assembly.c

```diff
@@ -476,6 +476,14 @@ mono_set_assemblies_path (const char* path)
 	}
 }
 
+void
+mono_set_assemblies_path_direct (char **path)
+{
+	if (assemblies_path)
+		g_strfreev (assemblies_path);
```

Comment:
is the `if` needed or just code bloat and optimizing the rare/half case?

Reply:
It's needed; that value can get set by other things (and we can't stop embedders from calling API functions multiple times regardless).

Reply:
The rest of initialize should also be made to not leak in that case really, but that can happen later.

Reply:
Like free, g_strfreev should be nop for null.

Reply:
Fair enough. I should make sure it actually does work that way first though.